web/profiles/custom/yalesites_profile/modules/custom/ys_toolbar/src/ToolbarItemsService.php

```diff
@@ -235,4 +253,54 @@ protected function buildButton(string $route, string $label): array {
     ];
   }
 
+  /**
+   * Build an off canvas item for the toolbar.
+   *
+   * @param string $route
+   *   The route for the toolbar item destination.
+   * @param string $label
+   *   The text label for the toolbar item.
+   *
+   * @return array
+   *   A rennder array for a toolbar item.
+   */
+  protected function buildOffCanvasButton(string $route, string $label): array {
+    return [
+      '#type' => 'toolbar_item',
+      'tab' => [
+        '#type' => 'link',
+        '#title' => $label,
+        '#url' => Url::fromRoute(
+          $route,
+          $this->redirectDestination->getAsArray(),
+        ),
+        '#access' => $this->accessManager->checkNamedRoute(
+          $route,
+          $this->getNodeRouteParams(),
+          $this->account
```

Comment:
I'm not sure this is what we want. The ability to update theme settings is note necessarily tied to one's permissions to update a node. What do you think about using the existing `yalesites manage settings` permission here? I admit it does not matter at the moment as there is only one real role. But I suspect in the future we will allow folks to edit content but not site settings.